polyfill/test/validStrings.mjs

```diff
@@ -461,11 +480,22 @@ function fuzzMode(mode) {
       const parsed = ES[`ParseTemporal${mode}String`](fuzzed);
       for (let prop of comparisonItems[mode]) {
         let expected = generatedData[prop];
-        if (prop !== 'tzName' && prop !== 'offset' && prop !== 'calendar') expected = expected || 0;
-        if (prop === 'offset' && expected) {
-          const parsedResult = ES.ParseDateTimeUTCOffset(parsed[prop]);
-          assert.equal(parsedResult.offsetNanoseconds, expected.offsetNanoseconds);
-          assert.equal(parsedResult.hasSubMinutePrecision, expected.hasSubMinutePrecision);
+        if (!['tzName', 'tzAnnotation', 'offset', 'offsetMinutes', 'calendar'].includes(prop)) {
+          expected = expected || 0;
+        }
+        if (mode === 'TimeZone') {
+          if (prop === 'offsetMinutes' && parsed.tzName) {
+            // An IANA name overrides the offset in an ISO string. If a name is
+            // present then ignore the offset.
+            expected = undefined;
+          }
+          if (prop === 'tzName' && expected === undefined && generatedData.z === 'Z') {
+            expected = 'UTC';
+          }
+        }
+        if (prop === 'offset') {
+          const parsedResult = parsed[prop] === undefined ? undefined : ES.ParseDateTimeUTCOffset(parsed[prop]);
+          assert.equal(parsedResult, expected, prop);
```

Comment:
I'd like to avoid putting logic in this part of the test as much as possible — it is supposed to test only the parsing. (I wonder if this change could be eliminated if the previous comment was addressed.)